kustomize-functions/validate-asm/image/main.go

```diff
@@ -127,14 +130,17 @@ func (filter) Filter(in []*yaml.RNode) ([]*yaml.RNode, error) {
 			}
 		}
 	}
+	if err := validateMinimumVCPUCount(totalVCPUCount); err != nil {
+		warningList = append(warningList, ValidationError{Warning, err})
```

Comment:
The ASM teams also have a couple discussions for a similar question - "should the auto installation generate errors on the vCPU requirements or should it just do a warning and continue?" 

Our answer for https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages/tree/master/scripts/asm-installer is - the install should give errors and helpful messages and stop there.

So to keep consistent behavior, I'd suggest we change kpt validate function to use errors instead of warnings.

Reply:
Okay. What about machine type check? From discussion in #66, we all agreed to return a warning for the node pool with unsatisfied machine type. But it seems to be a hard requirement from the [documentation](https://cloud.google.com/service-mesh/docs/gke-install-overview#requirements)
> Your GKE cluster must meet the following requirements: A machine type that has at least four vCPUs, such as e2-standard-4. If the machine type for your cluster doesn't have at least four vCPUs, change the machine type as described in Migrating workloads to different machine types.

Reply:
From memory, the last conversation we had was something like this: machine type and CPU count should be errors for now since that's what we document, and we don't want to validate something we don't support. Pretty soon the requirement will go way down anyways, so this won't be an issue for long.

Reply:
Fair enough. The validation was updated to return errors for both machine type and CPU count. PTAL.